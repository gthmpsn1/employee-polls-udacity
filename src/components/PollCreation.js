import { useState } from "react";
import { connect } from "react-redux";
import {useNavigate} from "react-router-dom";
import { handleCreatePoll } from "../actions/questions";

import Nav from "./Nav";

const PollCreation = (props) => {
    const navigate = useNavigate();
    const [textArea1, setTextArea1] = useState("");
    const [textArea2, setTextArea2] = useState("");
    const {user, dispatch} = props;

    const updateTextArea1 = (textArea1) => {
        setTextArea1(textArea1);
    };
    const updateTextArea2 = (textArea2) => {
        setTextArea2(textArea2);
    };

    const handleNewPoll = (e) => {
        e.preventDefault();
        const info = {
            textArea1,
            textArea2,
            user
        };
        dispatch(handleCreatePoll(info));
        setTextArea1("");
        setTextArea2("");
        navigate("/");
    } 

    return (
        <div className="poll-creation-container">
            <Nav />
            <div className="poll-creation-info">
                <h4>Create Your Own Poll</h4>
                <h3>Would you rather...</h3>
                <div className="option-create-container">
                    <form>
                        <label>Option #1</label>
                        <textarea 
                            maxLength="250" 
                            onChange={(e) => updateTextArea1(e.target.value)} 
                            value={textArea1} 
                            placeholder="Enter option #1" >
                        </textarea>
                        <p>{textArea1.length}/250</p>
                        <label>Option #2</label>
                        <textarea 
                            maxLength="250" 
                            onChange={(e) => updateTextArea2(e.target.value)} 
                            value={textArea2} 
                            placeholder="Enter option #2" >
                        </textarea>
                        <p>{textArea2.length}/250</p>
                        <input type="submit" onClick={handleNewPoll} value="Creat Poll" disabled={textArea1 === "" || textArea2 === ""} />
                    </form>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = ({authedUser, dispatch}) => ({
    dispatch: dispatch,
    user: authedUser
})

export default connect(mapStateToProps)(PollCreation);