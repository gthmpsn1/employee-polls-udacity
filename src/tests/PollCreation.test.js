/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from "@testing-library/react";
import middleware from "../middleware";
import reducer from "../reducers";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import PollCreation from "../components/PollCreation";

describe("Render PollCreation", () => {
    let mockStore = "";
    let view = "";
    const spy = jest.spyOn(global.console, "error");
    
    beforeEach(() => {
        mockStore = createStore(reducer, middleware);
        view = render(
            <MemoryRouter>
                <Provider store={ mockStore }>
                    <PollCreation />
                </Provider>
            </MemoryRouter>
        );
    });

    it("will render PollCreation component", async () => {
        await expect(view).toMatchSnapshot();
    });

    
    it("will return to dashboard when submitted ", async () => {
        const testTextArea1 = screen.queryByTestId("test-textArea1");
        fireEvent.change(testTextArea1, {target: {value: "option 1 text"}});
        const testTextArea2 = screen.queryByTestId("test-textArea2");
        fireEvent.change(testTextArea2, {target: {value: "option 2 text"}});
        const testSubmitPollButton = screen.queryByTestId("test-submit-poll-btn");
        fireEvent.click(testSubmitPollButton);

        const userUrl = window.location.pathname;
        expect(userUrl).toEqual(`/`);
        expect(spy).not.toHaveBeenCalled();
    });
});