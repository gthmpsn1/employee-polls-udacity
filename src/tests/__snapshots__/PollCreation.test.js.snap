// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render PollCreation will render PollCreation component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="poll-creation-container"
      >
        <nav
          class="nav"
        >
          <h1>
            Employee Polls 
            <span>
              by Gabriel Thompson
            </span>
          </h1>
          <ul>
            <div>
              <li>
                <a
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  href="/leaderboard"
                >
                  Leaderboard
                </a>
              </li>
              <li>
                <a
                  href="/new-poll"
                >
                  New Poll
                </a>
              </li>
            </div>
            <div>
              <li
                class="user-avatar"
                style="background-image: url(undefined);"
              />
              <li>
                <a
                  data-testid="TEST"
                  href="/"
                >
                  Logout, 
                </a>
              </li>
            </div>
          </ul>
        </nav>
        <div
          class="poll-creation-info"
        >
          <h4>
            Create Your Own Poll
          </h4>
          <h3>
            Would you rather...
          </h3>
          <div
            class="option-create-container"
          >
            <form>
              <label>
                Option #1
              </label>
              <textarea
                data-testid="test-textArea1"
                maxlength="250"
                placeholder="Enter option #1"
              />
              <p>
                0
                /250
              </p>
              <label>
                Option #2
              </label>
              <textarea
                data-testid="test-textArea2"
                maxlength="250"
                placeholder="Enter option #2"
              />
              <p>
                0
                /250
              </p>
              <input
                data-testid="test-submit-poll-btn"
                disabled=""
                type="submit"
                value="Creat Poll"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="poll-creation-container"
    >
      <nav
        class="nav"
      >
        <h1>
          Employee Polls 
          <span>
            by Gabriel Thompson
          </span>
        </h1>
        <ul>
          <div>
            <li>
              <a
                href="/"
              >
                Home
              </a>
            </li>
            <li>
              <a
                href="/leaderboard"
              >
                Leaderboard
              </a>
            </li>
            <li>
              <a
                href="/new-poll"
              >
                New Poll
              </a>
            </li>
          </div>
          <div>
            <li
              class="user-avatar"
              style="background-image: url(undefined);"
            />
            <li>
              <a
                data-testid="TEST"
                href="/"
              >
                Logout, 
              </a>
            </li>
          </div>
        </ul>
      </nav>
      <div
        class="poll-creation-info"
      >
        <h4>
          Create Your Own Poll
        </h4>
        <h3>
          Would you rather...
        </h3>
        <div
          class="option-create-container"
        >
          <form>
            <label>
              Option #1
            </label>
            <textarea
              data-testid="test-textArea1"
              maxlength="250"
              placeholder="Enter option #1"
            />
            <p>
              0
              /250
            </p>
            <label>
              Option #2
            </label>
            <textarea
              data-testid="test-textArea2"
              maxlength="250"
              placeholder="Enter option #2"
            />
            <p>
              0
              /250
            </p>
            <input
              data-testid="test-submit-poll-btn"
              disabled=""
              type="submit"
              value="Creat Poll"
            />
          </form>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
